packages/sample-plugin/webpack.config.ts

```diff
@@ -1,17 +1,37 @@
-/* eslint-disable @typescript-eslint/no-non-null-assertion */
 import path from 'path';
 import { DynamicRemotePlugin } from '@openshift/dynamic-plugin-sdk-webpack';
-import type { Configuration } from 'webpack';
+import { getSampleAppSharedModules } from '@sample-app/plugin-sdk-addons/shared-modules';
+import CSSMinimizerPlugin from 'css-minimizer-webpack-plugin';
+import type { Configuration, WebpackPluginInstance } from 'webpack';
+import { EnvironmentPlugin } from 'webpack';
 import extensions from './plugin-extensions';
 
+const isProd = process.env.NODE_ENV === 'production';
+
+const pathTo = (relativePath: string) => path.resolve(__dirname, relativePath);
+
+const sharedModules = getSampleAppSharedModules({
+  import: false,
```

Comment:
Same is approached when used with `singleton: true`. Although we should treat this as minimal app example so might be better to list all the dependencies we want in here and when bootstrapping the app. That way we can see directly which dependencies are shared. From this I'd have to go see the `getSampleAppSharedModules` function and read which dependencies are used. Not to speak about that the sample plugin requires `sample-app`

Reply:
@karelhala Are you suggesting we specify the (application provided) common shared modules in _both_ the application and the plugin?

The reason I've introduced `getSampleAppSharedModules` function is to have plugin use the same shared module definitions, reducing the risk of the plugin making wrong assumptions about shared modules.

If this information gets duplicated, assuming we want to add a new shared module to be provided by the host application, we'd have to update _all_ of the plugins :confused: (note that in Console plugin SDK, this information is hard-coded in `ConsoleRemotePlugin` which is application-specific, but our `DynamicRemotePlugin` is _not_ application-specific).

> Not to speak about that the sample plugin requires `sample-app`

Yeah, the idea was that there would be some package provided by the given host application (and published to npmjs) containing the common (application provided, singleton) shared module definitions. In the sample plugin, this use case is simulated via symlink dependency `link:../sample-app/src/plugin-sdk-addons`.

If there's a better way to share this information, I'm all for it.

Reply:
Yup, something like for minimal app config
```JS
{
 '@openshift/dynamic-plugin-sdk': { eager: true, singleton: true }, //required
 'react': { eager: true, singleton: true }, //required
 'react-dom': { eager: true, singleton: true }, //required
  'react-redux': { eager: true, singleton: true }, //required if using K8s hooks
  '@patternfly/react-core': { eager: true, singleton: true }, //optional
  '@patternfly/react-table': { eager: true, singleton: true }
}
```

And for plugin config
```
{
 '@openshift/dynamic-plugin-sdk': { import: false, singleton: true }, //required
 'react': { import: false, singleton: true }, //required
 'react-dom': { import: false, singleton: true }, //required
  'react-redux': { import: false, singleton: true }, //required if using K8s hooks
  '@patternfly/react-core': { import: false, singleton: true }, //optional
  '@patternfly/react-table': { import: false, singleton: true }
}
```

This will send direct message to those inspiring from this repository what they need to do and how it's achieved.